off-chain/src/Network/Types.purs

```diff
@@ -0,0 +1,207 @@
+module Seath.Network.Types where
+
+import Contract.Transaction (FinalizedTransaction)
+import Data.Either (Either)
+import Data.Generic.Rep (class Generic)
+import Data.Map (Map)
+import Data.Tuple.Nested (type (/\))
+import Data.Unit (Unit)
+import Effect.Aff (Aff)
+import Seath.Core.Types (UserAction)
+import Type.Function (type ($))
+
+-- TODO: replace this types with real ones.
+-- The names are indicatives but can change.
+type AsyncMutableQueueRef a = a
+type Ip = String
+type Port = String
+type ControlNumber = Int
+type QueueIndex = Int
+type ChainNumber = Int
+type ChainIndex = Int
+type RequestCounter = Int
+type MiliSeconds = Int
+type Time = Int
+type MutableInt = Int
+
+-- This represent a Map that can remember the order of insertion.
+-- it seems that we can use purescript-ordered-collections with a 
+-- type like this.
+-- I would like to use purescript-ordered-collections, but the 
+-- Ord instance it has is over the keys, and then we would need 
+-- to cast it to a plain map quite often.
+type OrderedMap keys values =
```

Comment:
> but the Ord instance it has is over the keys, and then we would need to cast it to a plain map quite often

Here is some [Haskell implementation](https://hackage.haskell.org/package/ordered-containers-0.2.3/docs/src/Data.Map.Ordered.Internal.html#OMap) I've found. Seems like it sacrifices some space, but avoids conversions.